src/preferences/listener.ts

```diff
@@ -19,16 +24,56 @@ export function createConfigChangeListener(): Disposable {
       const configs: WorkspaceConfiguration = workspace.getConfiguration();
 
       if (event.affectsConfiguration(SSL_PEM_PATHS)) {
+        // inform the sidecar that the SSL/TLS .pem paths have changed
         logger.debug(`"${SSL_PEM_PATHS}" config changed`);
         const pemPaths: string[] = configs.get(SSL_PEM_PATHS, []);
         await updatePreferences({
           tls_pem_paths: pemPaths,
         });
-      } else if (event.affectsConfiguration(SSL_VERIFY_SERVER_CERT_DISABLED)) {
+        return;
+      }
+
+      if (event.affectsConfiguration(SSL_VERIFY_SERVER_CERT_DISABLED)) {
+        // inform the sidecar that the server cert verification has changed
         logger.debug(`"${SSL_VERIFY_SERVER_CERT_DISABLED}" config changed`);
         // if the user disables server cert verification, trust all certs
         const trustAllCerts: boolean = configs.get(SSL_VERIFY_SERVER_CERT_DISABLED, false);
         await updatePreferences({ trust_all_certificates: trustAllCerts });
+        return;
+      }
+
+      // --- PREVIEW SETTINGS --
+      // Remove the sections below once the behavior is enabled by default and a setting is no
```

Comment:
There's a lot of notes about removing things... not a blocker, but I wonder if in the future we could isolate this logic and have a shorter way of opting these preview features in?

Reply:
Agreed! This is still somewhat new to me/us, so having a more generalized pattern for changing behavior based on preview settings would be nice if we continue to use them